src/Workspaces/Core/Portable/Rename/RenameLocation.ReferenceProcessing.cs

```diff
@@ -276,25 +276,29 @@ private static string TrimNameToAfterLastDot(string name)
                 }
 
                 // If we're renaming a named type, we'll also have to find constructors and
-                // destructors declarations that match the name
+                // destructors declarations that match the name if we're in C#. In VB these
+                // methods are not named to match the class.
                 if (referencedSymbol.Kind == SymbolKind.NamedType)
                 {
                     var namedType = (INamedTypeSymbol)referencedSymbol;
-
-                    foreach (var method in namedType.GetMembers().OfType<IMethodSymbol>())
+                    if (namedType.Language == LanguageNames.CSharp)
```

Comment:
Given talk is beginning about how to refactor this service to support other languages, this might just be creating debt there.

There's already other code in Rename to block renaming from a VB constructor (which you can't do) but would be allowed for a C# constructor. How does that work?